doc/nrf/protocols/matter/getting_started/custom_clusters.rst

```diff
@@ -0,0 +1,408 @@
+
+   .. code-block::
+
+      west zap-generate --full
+
+After completing these steps, the following changes will be visible within your sample directory:
+
+* The new cluster description file :file:`MyCluster.xml`.
+* The updated data model definition file :file:`zcl.json` with the new cluster and relative paths to the Matter data model directory.
+* The generated C++ source files for the new cluster.
+* The updated :file:`.zap` file with the new cluster configuration and relative path to the :file:`zcl.json` file.
+
+Once the new cluster is added to the Matter application, you can call the ``zap-gui`` command without the additional ``--clusters`` argument.
+However, you still need to provide the path to the :file:`zcl.json` file if you created a new one in a different location than the default one.
+
+
+.. rst-class:: numbered-step
+
+Align cmake configuration with the new cluster
+**********************************************
+
+Generating zap files with ``--full`` option creates new source files under :file:`zap-generated/app-common` that need to override the default files located in the Matter SDK in :file:`zzz_generated/app-common` directory.
+To override the path, you need to set the ``CHIP_APP_ZAP_DIR`` variable in the CMakeLists.txt file, pointing to the parent of generated :file:`app-common` directory, before initializing the Matter Data Model.
+
+As custom clusters are not part of the default Matter SDK, you need to additionally pass a list of all new cluster names in an ``EXTERNAL_CLUSTERS`` argument when calling ``chip_configure_data_model``.
+
+For example, the following code snippet shows how to modify the Matter template :file:`CMakeLists.txt` file with the new cluster:
+
+   .. code-block:: cmake
+
+      # Override zap-generated directory.
+      get_filename_component(CHIP_APP_ZAP_DIR ${CONFIG_NCS_SAMPLE_MATTER_ZAP_FILES_PATH}/zap-generated REALPATH CACHE)
+
+      # Enable GNU STD support and initialize the Matter Data Model.
+      include(${ZEPHYR_CONNECTEDHOMEIP_MODULE_DIR}/config/nrfconnect/app/enable-gnu-std.cmake)
+      include(${ZEPHYR_CONNECTEDHOMEIP_MODULE_DIR}/src/app/chip_data_model.cmake)
+
+      # NORDIC SDK APP START
+
+      # Include all source files that are located in the Matter common directory.
+      include(${ZEPHYR_NRF_MODULE_DIR}/samples/matter/common/cmake/source_common.cmake)
+
+      target_include_directories(app PRIVATE
+          src
+          ${CONFIG_NCS_SAMPLE_MATTER_ZAP_FILES_PATH}
+      )
+
+      target_sources(app PRIVATE
+          src/app_task.cpp
+          src/main.cpp
+      )
+
+      chip_configure_data_model(app
+          INCLUDE_SERVER
+          BYPASS_IDL
+          GEN_DIR ${CONFIG_NCS_SAMPLE_MATTER_ZAP_FILES_PATH}/zap-generated
+          ZAP_FILE ${CMAKE_CURRENT_SOURCE_DIR}/${CONFIG_NCS_SAMPLE_MATTER_ZAP_FILES_PATH}/template.zap
+          EXTERNAL_CLUSTERS "MY_NEW_CLUSTER" # Add EXTERNAL_CLUSTERS flag
+      )
+
```

Comment:
```suggestion

/* Existing code in CMakeList.txt */

      chip_configure_data_model(app
          INCLUDE_SERVER
          BYPASS_IDL
          GEN_DIR ${CONFIG_NCS_SAMPLE_MATTER_ZAP_FILES_PATH}/zap-generated
          ZAP_FILE ${CMAKE_CURRENT_SOURCE_DIR}/${CONFIG_NCS_SAMPLE_MATTER_ZAP_FILES_PATH}/template.zap
          EXTERNAL_CLUSTERS "MY_NEW_CLUSTER" # Add EXTERNAL_CLUSTERS flag
      )

```

To focus on the actual change?

Reply:
ok, added `project(matter-template)` line at the beginning to show where CHIP_APP_ZAP_DIR should be set